core/types/header_ext_test.go

```diff
@@ -0,0 +1,146 @@
+// (c) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package types
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"math/big"
+	"reflect"
+	"testing"
+
+	"github.com/ava-labs/libevm/common"
+	ethtypes "github.com/ava-labs/libevm/core/types"
+	"github.com/ava-labs/libevm/rlp"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHeaderRLP(t *testing.T) {
+	t.Parallel()
+
+	got := testHeaderEncodings(t, rlp.EncodeToBytes, rlp.DecodeBytes)
+
+	// Golden data from original coreth implementation, before integration of
+	// libevm. WARNING: changing these values can break backwards compatibility
+	// with extreme consequences as block-hash calculation may break.
+	const (
+		wantHex     = "f90234a00100000000000000000000000000000000000000000000000000000000000000a00200000000000000000000000000000000000000000000000000000000000000940300000000000000000000000000000000000000a00400000000000000000000000000000000000000000000000000000000000000a00500000000000000000000000000000000000000000000000000000000000000a00600000000000000000000000000000000000000000000000000000000000000b901000700000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000008090a0b0c0da00e00000000000000000000000000000000000000000000000000000000000000880f00000000000000a015000000000000000000000000000000000000000000000000000000000000001016171213a01400000000000000000000000000000000000000000000000000000000000000"
+		wantHashHex = "7b63c1c763cc1527f12de7e5331b92d66ea201d1e6d7cd65a79bf8e3b99c104d"
+	)
+
+	assert.Equal(t, wantHex, hex.EncodeToString(got), "Header RLP")
+
+	header, _ := headerWithNonZeroFields()
+	gotHashHex := header.Hash().Hex()
+	assert.Equal(t, "0x"+wantHashHex, gotHashHex, "Header.Hash()")
+}
+
+func TestHeaderJSON(t *testing.T) {
+	t.Parallel()
+
+	testHeaderEncodings(t, json.Marshal, json.Unmarshal)
+}
+
+func testHeaderEncodings(t *testing.T, encode func(any) ([]byte, error), decode func([]byte, any) error) []byte {
+	t.Helper()
+
+	input, _ := headerWithNonZeroFields() // the Header carries the HeaderExtra so we can ignore it
+	b, err := encode(input)
```

Comment:
nit: could we call this `bytes`?

Reply:
I originally called it `buf`, which is a common name in the standard lib for a buffer of bytes, but am happy with another word. I agree that `b` is too short for a variable used over 15 lines with a lot else going on.

Reply:
Renamed to `encoded` (named the return as well because the signature of the test helper wasn't really helpful knowing what bytes were returned)